lib/max_subarray.py

```diff
@@ -2,11 +2,14 @@
 def max_sub_array(nums):
     """ Returns the max subarray of the given list of numbers.
         Returns 0 if  nums is None or an empty list.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
     if nums == None:
         return 0
     if len(nums) == 0:
         return 0
-    pass
+
+    for i in range(1, len(nums)):
+        nums[i] = max(nums[i-1]+nums[i], nums[i])
+    return max(nums)
```

Comment:
You can do this without a list.  Instead you can use 2 local variables, one for the local max subarray and the other for the largest subarray sum found.